src/static_canvas.class.js

```diff
@@ -625,9 +625,9 @@
      * @param {fabric.Object} obj Object that was removed
      */
     _onObjectRemoved: function(obj) {
+      obj._set('canvas', undefined);
       this.fire('object:removed', { target: obj });
       obj.fire('removed', { target: this });
-      obj._set('canvas', undefined);
```

Comment:
is there any bug with removing the canvas before or after?
We could pass a reference to the canvas in the event itself as oldCanvas?
In general what are the reason for this line swap?

Reply:
https://github.com/fabricjs/fabric.js/pull/7859#issuecomment-1109377515

In terms of design and naming it's `removed` so I think canvas should not be referenced anymore once the event triggers (this is how group behaves) so it is consistent across fabric

Reply:
And it is passed as the target

Reply:
oh right this is the canvas not the object. i didn't think of it.